src/lighteval/models/utils.py

```diff
@@ -41,17 +41,18 @@ def _get_dtype(dtype: Union[str, torch.dtype], config: Optional[AutoConfig] = No
         torch.dtype: The torch dtype based on the input arguments.
     """
 
-    if config is not None:  # For quantized models
-        if hasattr(config, "quantization_config"):
-            _torch_dtype = None  # must be inferred
-        else:
-            _torch_dtype = config.torch_dtype
-    elif isinstance(dtype, str) and dtype not in ["auto", "4bit", "8bit"]:
+    if config is not None and hasattr(config, "quantization_config"):
+        # must be infered
+        return None
+
+    if isinstance(dtype, str) and dtype not in ["auto", "4bit", "8bit"]:
```

Comment:
If the goal is to overwrite the possibly existing quantization config then, I think this condition
1. Should be placed first in the function logic
2. Should be divided in 2 sections
```
   if isinstance(dtype, str):
       if dtype not in ["auto", "4bit", "8bit"]:
           return dtype
       else:
           return getattr(torch, dtype)
```

Reply:
if not, the next condition checks might overwrite this param instead

Reply:
I we have a quantization config in the model congo then we do not want to overwrite it and we return None.
else if we have a dtype that is not a quant type then we return it.

Reply:
agreed yes